shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -76,18 +76,30 @@ @interface FlutterTextRange : UITextRange<NSCopying>
 
 + (instancetype)rangeWithNSRange:(NSRange)range;
 
++ (instancetype)rangeWithNSRange:(NSRange)range withText:(NSString*)text;
+
 @end
 
 @implementation FlutterTextRange
 
 + (instancetype)rangeWithNSRange:(NSRange)range {
-  return [[[FlutterTextRange alloc] initWithNSRange:range] autorelease];
+  return [[[FlutterTextRange alloc] initWithNSRange:range withText:nil] autorelease];
+}
+
++ (instancetype)rangeWithNSRange:(NSRange)range withText:(NSString*)text {
```

Comment:
Given that the text itself isn't part for the state, `forText:` seems like the better name.

Reply:
Deleted